### PR TITLE
[DPE-2109] Integrate Spark History Server charm with traefik-k8s

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -69,6 +73,18 @@ def spark_conf(self):
         """Return the dict representation of the configuration file."""
         s3_log_dir = self.s3_log_dir
         conn_config = self.s3_creds_client.get_s3_connection_info()
+
+        ingress_pattern = re.compile(r"http://.*?/")
```
Comment (welpaolo) [inline src/config.py:77]: As we discussed, we need to understand how to properly handle validation of this field.

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -189,3 +189,46 @@ async def test_build_and_deploy(ops_test: OpsTest):
...
+@pytest.mark.abort_on_fail
+async def test_ingress(ops_test: OpsTest):
+    """Build the charm-under-test and deploy it together with related charms.
+
+    Assert on the unit status before any relations/configurations take place.
+    """
+    # Deploy the charm and wait for waiting status
+    _ = await ops_test.model.deploy(
+        INGRESS_CHARM,
+        channel="edge",
```
Comment (welpaolo) [inline tests/integration/test_charm.py:203]: I think we can put channel and series values into costants.

**Review by welpaolo — APPROVED** [review]
Looks good to me!

I just left some minor comments.

File: src/charm.py [hunk 3]
```diff
@@ -78,6 +97,8 @@ def apply_s3_credentials(self) -> str:
         """Apply s3 credentials to container."""
         container = self.unit.get_container(CONTAINER)
 
+        self.logger.info(f"Configuration file: {self.spark_config.contents}")
```
[the remark this answers is not in the document]
Comment (deusebio) [reply src/charm.py:100]: Yes, it was useful for debugging and implementing, but it is probably best to get rid of this

**Review by marcoppenheimer — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -36,6 +36,8 @@ jobs:
       - name: Install tox
         # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
         run: python3 -m pip install tox
+      - name: Run tests
+        run: tox run -e fmt
```
[the remark this answers is not in the document]
Comment (deusebio) [reply .github/workflows/ci.yaml:40]: I have been working with the [upload-charm-docs](https://github.com/canonical/upload-charm-docs/blob/main/tox.ini#L25) project too often lately :D 

Definitely reverting back, I really thought that we were using - as DataPlatform - fmt. Great catch!

**Pull request merged**